tools/report-converter/codechecker_report_converter/sparse/output_parser.py

```diff
@@ -0,0 +1,91 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+
+import logging
+import os
+import re
+
+from ..output_parser import Message, Event, BaseParser
+
+LOG = logging.getLogger('ReportConverter')
+
+
+class SparserParser(BaseParser):
+    """ Parser for sparse console output. """
+
+    def __init__(self):
+        super(SparserParser, self).__init__()
+
+        # Regex for parsing a sparse message.
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':' and a space.
+            r'(?P<column>\d+?): '
+            # Severity followed by a ':'.
+            r'(?P<severity>(error|warning)):'
+            # Checker message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+        # Match a sparse detailed info
+        self.message_detailed_info = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':' , 4 spaces (not a tab because a 
+            # tab has 7 spaces)
+            r'(?P<column>\d+?):    '
+            # Detailed info type (expected | got) followed by a ':'.
+            r'(?P<severity>(expected|got|struct)) '
```

Comment:
I think this could be the part of the "message" part, so we shouldn't distinguish these words separately as "severity". These are just the first words of the message. I don't know if it's guaranteed that one of these is the first word of the message.

Reply:
Actually these words are keywords which distinguish the lines from the normal messages while using regex. They are combined with the message again when creating the Note.

Reply:
severity is still the wrong word above, right? It is probably something like, message_extension, extension, just "more", or so. I hope you still figure out how we can display multi-line warnings properly in CodeChecker. At the moment, the message is where but it looks pretty ugly and "broken".